### DISCUSSION TITLE
Prevent creation of duplicate events - expo-calendar

### DISCUSSION THREAD

**Discussion by anant-nft** [expo/expo#18435] [Archived: Questions]
I am creating events using `createEventAsync` method. But if I run the code again, it creates duplicate events. How to prevent this ? Already tried checking if event already exist as below - 

```
function eventExists(newEventData) {
   const existingEvents = await Calendar.getEventsAsync([calendarId], startDate, endDate)
   return existingEvents.some(existingEvent => 
      existingEvent.id === newEventData.id
    )
}
```

But this doesn't work because existing event's id is different from new event data's internal id.

**Comment by barthap** [comment]
This is not a perfect solution, but rather a workaround and depends on use case, but maybe instead of comparing IDs, compare some other fields, like name, start/end date etc.

**Reply by anant-nft** [reply]
Currently using a similar workaround using event notes.
